Migrate recipe to rattler-build v1 format

vicentebolea opened the pull request:
This PR updates the adios2-feedstock to use the new rattler-build v1
recipe format as introduced by CEP 13.

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I was trying to look for recipes to lint for you, but it appears we have a merge conflict. Please try to merge or rebase with the base branch to resolve this conflict.

Please ping the 'conda-forge/core' team (using the `@` notation in a comment) if you believe this is a bug.

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/recipe.yaml```) and found it was in an excellent condition.

vicentebolea commented:
Thanks @ax3l here is the feature that we needed from experimental: https://rattler-build.prefix.dev/dev/multiple_output_cache/

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I was trying to look for recipes to lint for you, but it appears we have a merge conflict. Please try to merge or rebase with the base branch to resolve this conflict.

Please ping the 'conda-forge/core' team (using the `@` notation in a comment) if you believe this is a bug.

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/recipe.yaml```) and found it was in an excellent condition.

vicentebolea commented:
@ax3l CI finally passing, since it is imposible to share a cache between the different pacakges (experimental) and making two builds per jobs seems overkill, I decided to make libadios2 a alias package of adios2 in this manner users depending on libadios2 will not be affected (other than having a larger dependency) which is really a problem in only two projects across the forge:

https://github.com/search?q=org%3Aconda-forge+path%3Arecipe%2Frecipe.yaml+OR+path%3Arecipe%2Fmeta.yaml++NOT+is%3Afork+NOT+repo%3Aconda-forge%2Fadios2-feedstock+libadios2+&type=code

conda-forge-admin commented:
Hi! This is the friendly conda-forge automerge bot!

Commits were made to this PR after the `automerge` label was added. For security reasons, I have disabled automerge by removing the `automerge` label. Please add the `automerge` label again (or ask a maintainer to do so) if you'd like to enable automerge again!

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/recipe.yaml```) and found it was in an excellent condition.

I do have some suggestions for making it better though...


For **recipe/recipe.yaml**:

 * ℹ️ It looks like the '???' output doesn't have any tests.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/24534631719](https://github.com/conda-forge/conda-forge-webservices/actions/runs/24534631719). Examine the logs at this URL for more detail.</sub>

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/recipe.yaml```) and found it was in an excellent condition.

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/recipe.yaml```) and found it was in an excellent condition.

I do have some suggestions for making it better though...


For **recipe/recipe.yaml**:

 * ℹ️ It looks like the '???' output doesn't have any tests.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/24596803099](https://github.com/conda-forge/conda-forge-webservices/actions/runs/24596803099). Examine the logs at this URL for more detail.</sub>

vicentebolea commented:
waiting on: https://github.com/prefix-dev/rattler-build/pull/2432

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/recipe.yaml```) and found it was in an excellent condition.

vicentebolea committed:
REFACTOR: Migrate recipe to rattler-build v1 format with shared cmake staging build

Use a staging output to run cmake once and split installed files between
libadios2 (C++ libs + headers) and adios2 (Python bindings + CLI tools).
Fixes osx-64 DILL/libffi segfault with DILL_NATIVE_ONLY=ON.

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I failed to even lint the recipe, probably because of a conda-smithy bug :cry:. This likely indicates a problem in your `meta.yaml`, though. To get a traceback to help figure out what's going on, install conda-smithy and run `conda smithy recipe-lint --conda-forge .` from the recipe directory. You can also examine the [workflow logs](https://github.com/conda-forge/conda-forge-webservices/actions/runs/24791027896) for more detail.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/24791027896](https://github.com/conda-forge/conda-forge-webservices/actions/runs/24791027896). Examine the logs at this URL for more detail.</sub>

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/recipe.yaml```) and found it was in an excellent condition.

vicentebolea committed:
MNT: Re-rendered with conda-smithy 3.61.2 and conda-forge-pinning 2026.04.28.00.10.39

Other tools:
- conda-build 26.3.0
- rattler-build 0.63.1
- rattler-build-conda-compat 1.4.14

vicentebolea committed:
REFACTOR: Remove unneeded deps (gnuconfig, fortran+ninja in test)

conda-forge-admin commented:
Hi! This is the friendly conda-forge automerge bot!

I considered the following status checks when analyzing this PR:
 - **linter**: passed
 - **azure**: passed
 - **github-actions**: passed


Thus the PR was passing and merged! Have a great day!
